express-post-json/index.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res, next) => {
+  const gradesArr = [];
+  for (const property in grades) {
+    gradesArr.push(grades[property]);
+  }
+  res.json(gradesArr);
+  next();
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  if (req.method !== 'POST') {
+    res.status(400).send('bad request');
+    return;
+  }
+  grades[nextId] = req.body;
+  grades[nextId].id = nextId;
+  nextId++;
+  res.sendStatus(201);
```

Comment:
Here we want to send the newly created grade back to the client. So you need to save the new grade into a variable making it easy to send back to the client. For example:

```JAVASCRIPT
const newGrade = req.body;
const id = nextId;
nextId++;
newGrade.id = id;
grades[id] = newGrade;
res.status(201).json(newGrade);
```